src/Controller/ResourceController.php

```diff
@@ -345,27 +364,41 @@ private function getDirectory(Request $request, $path, $responseFormat, array $v
             ];
         }
 
-        $content = $this->getGraphForPath($request, $path)->serialise($responseFormat, $options);
+        $content = $this->getGraphForPath($request, $requested_path)->serialise($responseFormat, $options);
 
         $headers = [
-            "Last-Modified" => $modifiedTime->format(\DateTime::W3C),
-            "Link" => ["<{$this->LDP_NS}Resource>; rel=\"type\"",
-                       "<{$this->LDP_NS}BasicContainer>; rel=\"type\""],
-            "Vary" => "Accept",
+            "Link" => ["<".self::LDP_NS."Resource>; rel=\"type\"",
+                       "<".self::LDP_NS."BasicContainer>; rel=\"type\""],
             "Content-Type" => $responseMimeType,
             "Content-Length" => strlen($content)
         ];
 
-        if (!$doGet) {
-            $content = '';
+        if ($request->getMethod() == 'GET') {
+            $response->setContent($content);
         }
-        return new Response($content, 200, $headers);
+        $response->headers->add($headers);
+        return $response;
+    }
+
+    /**
+     * Simple Convulsive Etag generator for files
+     *
+     * @return array
+     */
+    private function provideEtag()
```

Comment:
@DiegoPino it's probably more a style thing, but I generally try to keep class properties immutable (i.e. set by the constructor) and make methods more like pure functions. But we can clean that up later.